drivers/java/src/main/java/com/rethinkdb/net/Connection.java

```diff
@@ -345,7 +345,7 @@ public void runNoReply(ReqlAst term, OptArgs globalOpts) {
 
     void stop(Cursor cursor) {
         // The server does reply to a stop request, though the response doesn't have a value.
```

Comment:
Thanks first of all for your research on this and the bug fix @QianJin2013 !
My only suggestion is changing this comment to state something like this:
```
// While the server does reply to the stop request, we ignore that reply.
// This works because the response pump in `connect` ignores replies for which
// no waiter exists.
```